CONTRIBUTING.md

```diff
@@ -0,0 +1,41 @@
+- We ask you only to pick issues for which you are confident that you have a solution;
+- If you are a newcomer to the repo, create a personal fork and clone it locally. Check out a new branch with a name relevant to the issue you are working on;
+- Follow the coding style of the area you are editing, including indentation, and be consistent with quotation marks and spacing;
+- Use no higher than [ECMAScript 5](https://caniuse.com/es5) - notably, do not use arrow functions or `async` functions. However, Promises *are*
+  supported via a polyfill;
+- Do not prettify code you are not working on;
+- You must test your code yourself before asking for review, like this:
+  - set up a local Web server (we recommend Node/NPM's [http-server](https://www.npmjs.com/package/http-server));
+  - serve the root directory of the repository (e.g. `http-server .`);
+  - in a browser, navigate to the URL of the main `index.html` (e.g. http://localhost:8080/www/index.html);
+  - manually test your fix in at least Firefox and Chromium (Edge or Chrome), ideally also in IE11 or in "IE Mode" in Edge;
+  - be sure to test your fix in both "JQuery" mode and "Service Worker" mode (see Configuration);
+  - run the Unit tests (see below) in at least the above browsers.
+
+If all the tests are working fine, you can finally test the extension versions, like this:
+
+  - Remove the '-WIP' from the version key from the manifest.json file present in the root of this repo;
+  - In Chromium, you can install the extension by loading the root folder with Extensions -> Load Unpacked (with Developer Mode turned ON) -> select the root folder of the repository;
+  - In Firefox, you can load an extension with Manage Your Extensions -> Debug Add-ons -> Load Temporary Add-on, and then pick any file in the repository.
+
+If your feature works and tests are passing, make a PR, describe the testing you have done, and ask for a code review.
+
+Please note that the app caches its own code so that it can run as an offline-first Progressive Web App. This can complicate development, because you
+may not see your changes, even after you refresh the browser. In Configuration, under "Expert settings", you will find a button that allows you to do
+a full app reset, which will erase the PWA. When Service Worker mode is turned on, there is also a checkbox that bypasses the App Cache. You can turn
+this on if you are frequently changing code and refreshing. Remember to turn it off for final testing. You can manually delete the App Cache in
+the browser's DevTools (see Application or Storage tabs). We also recommend you disable the browser's built-in cache, using the checkbox in the DevTools Network tab.
+
+## Unit tests
+
+Unit tests can be run by opening `tests/index.html` file in Firefox, Edge, or Chromium/Chrome.
+
+Before running the tests, a one-time set up is needed to fetch development dependencies from the npm registry. Run `npm ci --ignore-scripts` to fetch the same versions as we use in CI.
```

Comment:
@gaurav7019 There's a revised version of the Unit tests section here:

https://github.com/kiwix/kiwix-js/pull/792/files

Could you kindly use that version, as some things have changed and been expanded?

Reply:
Once you've transferred that text, I'll remove mine.

Reply:
Done